source/s3_auto_ranged_put.c

```diff
@@ -518,24 +508,17 @@ static bool s_s3_auto_ranged_put_update(
             }
 
             bool should_create_next_part_request = false;
+            bool request_uploaded = false;
             if (auto_ranged_put->has_content_length && (auto_ranged_put->synced_data.num_parts_started <
                                                         auto_ranged_put->total_num_parts_from_content_length)) {
 
-                /* Check if the etag/checksum list has the result already */
-                for (size_t part_index = auto_ranged_put->threaded_update_data.next_part_number - 1;
-                     part_index < aws_array_list_length(&auto_ranged_put->synced_data.part_list);
-                     ++part_index) {
-
-                    struct aws_s3_mpu_part_info *part;
-                    aws_array_list_get_at(&auto_ranged_put->synced_data.part_list, &part, part_index);
-                    if (part != NULL) {
-                        /* part already downloaded, skip it here and prepare will take care of adjusting the buffer */
-                        ++auto_ranged_put->threaded_update_data.next_part_number;
+                size_t part_index = auto_ranged_put->threaded_update_data.next_part_number - 1;
 
-                    } else {
-                        // incomplete part found. break out and create request for it.
-                        break;
-                    }
+                struct aws_s3_mpu_part_info *part = NULL;
```

Comment:
minor: maybe add field to part info to indicate that we got info for that part from list parts, i.e. it was previously uploaded. relying on part being present in the list seems a bit fragile of a logic